### PR TITLE
Image and Mask can accept PIL images

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:

- Fixes "make Image(some_pil_image)" from TODO list (https://github.com/pytorch/vision/issues/7217)
- Added tests
- Updated dataset wrapper
- Increase tolerence for flaky cpu-bbox-rotation eager vs scripted tests

### CODE REVIEW

File: torchvision/prototype/datapoints/_image.py [hunk 1]
```diff
@@ -23,6 +23,11 @@ def __new__(
         device: Optional[Union[torch.device, str, int]] = None,
         requires_grad: Optional[bool] = None,
     ) -> Image:
+        if isinstance(data, PIL.Image.Image):
+            from ..transforms import functional as F
```
Comment (NicolasHug) [inline torchvision/prototype/datapoints/_image.py:27]: isn't this also accessible `self._F`?

Comment (pmeier) [inline torchvision/prototype/datapoints/_image.py:27]: Nit: I dislike the relative imports over the boundaries of large namespaces

```suggestion
            from torchvision.prototype.transforms import functional as F
```

File: test/test_prototype_datapoints.py [hunk 2]
```diff
@@ -130,3 +132,24 @@ def test_wrap_like():
...
+@pytest.mark.parametrize("data", [torch.randint(0, 10, size=(1, 32, 32)), Image.new("L", (32, 32), color=2)])
+def test_mask_instance(data):
+    mask = datapoints.Mask(data)
+    assert isinstance(mask, torch.Tensor)
+    assert mask.ndim == 3 and mask.shape[0] == 1
+
+
+@pytest.mark.parametrize("data", [torch.randint(0, 32, size=(5, 4)), [[0, 0, 5, 5], [2, 2, 7, 7]]])
+def test_bbox_instance(data):
+    bboxes = datapoints.BoundingBox(data, format="XYXY", spatial_size=(32, 32))
```
Comment (pmeier) [inline test/test_prototype_datapoints.py:153]: Can we parametrize over enum and str for the format here?

**Review by NicolasHug — APPROVED** [review]
Thank you @vfdev-5 !
Changes LGTM if green, let's just wait for @pmeier's input as well

File: torchvision/prototype/datapoints/_image.py (hunk 1 above)
Comment (pmeier) [reply torchvision/prototype/datapoints/_image.py:27]: In theory yes, but inside `__new__` we don't have `self` yet.

Comment (vfdev-5) [reply torchvision/prototype/datapoints/_image.py:27]: It's a classmethod and even `__F` can be None if this is the first call, so I prefered to import

File: test/prototype_transforms_kernel_infos.py [hunk 3]
```diff
@@ -854,7 +854,7 @@ def sample_inputs_rotate_video():
             F.rotate_bounding_box,
             sample_inputs_fn=sample_inputs_rotate_bounding_box,
             closeness_kwargs={
-                **scripted_vs_eager_double_pixel_difference("cpu", atol=1e-6, rtol=1e-6),
```
Comment (pmeier) [inline test/prototype_transforms_kernel_infos.py:857]: Is this a driveby?

**Review by pmeier — APPROVED** [review]
Thanks Victor!

File: test/prototype_transforms_kernel_infos.py (hunk 3 above)
Comment (vfdev-5) [reply test/prototype_transforms_kernel_infos.py:857]: Yes, 

> Increase tolerence for flaky cpu-bbox-rotation eager vs scripted tests

as they can fail in a flaky way:
- https://github.com/pytorch/vision/actions/runs/4164786358/jobs/7206896467
- https://github.com/pytorch/vision/actions/runs/4164786358/jobs/7206895696
